conda/dev.yaml

```diff
@@ -8,6 +8,4 @@ dependencies:
   - docker-compose
   - celery
   - make
-  - pip:
-    - git+https://github.com/luabida/satellite-weather-downloader.git@xarray-refactor #TODO: remove it when bumping version
-    # - satellite-weather-downloader
+  - poetry >=1.2
```

Comment:
it is necessary for the poetry install command
don't install that using conda env:
> git+https://github.com/luabida/satellite-weather-downloader.git@xarray-refactor

Reply:
the container will use this repo as a package, that will come with the necessary deps https://pypi.org/project/satellite-weather-downloader/

Reply:
let's break it down in order to understand what we have here.

1. if you want to test the package on docker on CI, so you need to have it from the repo ... not from the pypi. 
2. if you want to have a docker image with the package from pypi, it doesn't make sense to have it on CI for PR or push .. you will need to have it on workflow_dispatch.

so that is the common workflow .. if you have a very specific workflow you need first check if that really makes sense .. very custom workflow could led to difficult maintenance ... 

but if you are not using poetry on docker .. ensure you have tests for that on CI .. otherwise doesn't make sense to  support poetry.

keep in mind that docker for development and for production are or could be different ... so if you want to test your library on docker on ci ... you should use it from your repo .. not from pypi ... and if you want that for production .. you should use it on workflow dispatch

Reply:
There is basically two things in this repo, the conda deps are only for the containers, because it only needs celery in order to run tasks.
The pyproject.toml is for and updating this package on PYPI: https://pypi.org/project/satellite-weather-downloader/, then the tasks will import this package to use its modules as normal python imports. The main purpose of doing this way is that you can install `satellite-weather-downloader` to use outside a docker environment.

Reply:
not sure if you understood my points ... I recommend you to revisit that and check that again

Reply:
ping me if you want a call to understand better that.